src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1217,6 +1221,25 @@ public static void applySslContext()
         }
     }
 
+    public static void applyCryptoProvider()
+    {
+        try
+        {
+            if (conf.crypto_provider == null)
+                conf.crypto_provider = new ParameterizedClass("org.apache.cassandra.security.DefaultCryptoProvider", null);
```

Comment:
@ayushisingh29 would it be possible to use 'DefaultCryptoProvider.class.getName()' instead of putting there just string? What if the name of that provider changes or the package it is located in changes? Then you would need to fix it here as well ... Using class.getName(), you have it for free.

Does it interfere with DatabaseDescriptorRefTest or something similar that it can not be used like that?

Reply:
With the use of  'DefaultCryptoProvider.class.getName()' I have to add DefaultCryptoProvider to DatabaseDescriptorRefTest, which I do not see as an issue. It is better to use class.getName()